app/internal/notification.py

```diff
@@ -0,0 +1,174 @@
+    """
+    return notification.status in ARCHIVED
+
+
+def is_owner(user, notification: NOTIFICATION_TYPE) -> bool:
+    """Checks if user is owner of the notification.
+
+    Args:
+        notification: a NOTIFICATION_TYPE object.
+        user: user schema object.
+
+    Returns:
+        True or raises HTTPException.
+    """
+    if notification.recipient_id == user.user_id:
+        return True
+
+    msg = "The notification you are trying to access is not yours."
+    raise HTTPException(
+        status_code=HTTP_401_UNAUTHORIZED,
+        detail=msg,
+    )
+
+
+def raise_wrong_id_error() -> None:
+    """Raises HTTPException.
+
+    Returns:
+        None
+    """
+    raise HTTPException(
+        status_code=HTTP_406_NOT_ACCEPTABLE,
+        detail=WRONG_NOTIFICATION_ID,
+    )
+
+
+def filter_notifications(
+    session: Session,
+    user_id: int,
+    func: Union[_is_unread, _is_archived],
+) -> List[NOTIFICATION_TYPE]:
+    """Filters notifications by "func"."""
+    return list(filter(func, get_all_notifications(session, user_id)))
+
+
+def get_unread_notifications(
+    session: Session,
+    user_id: int,
+) -> List[NOTIFICATION_TYPE]:
+    """Returns all unread notifications."""
+    return filter_notifications(session, user_id, _is_unread)
+
+
+def get_archived_notifications(
+    session: Session,
+    user_id: int,
+) -> List[NOTIFICATION_TYPE]:
+    """Returns all archived notifications."""
+    return filter_notifications(session, user_id, _is_archived)
+
+
+def get_all_notifications(
+    session: Session,
+    user_id: int,
+) -> List[NOTIFICATION_TYPE]:
+    """Returns all notifications."""
+    invitations: List[Invitation] = get_all_invitations(
+        session,
+        recipient_id=user_id,
+    )
+    messages: List[Message] = get_all_messages(session, user_id)
+
+    notifications = invitations + messages
+    return sort_notifications(notifications)
+
+
+def sort_notifications(
+    notification: List[NOTIFICATION_TYPE],
+) -> List[NOTIFICATION_TYPE]:
+    """Sorts the notifications by the creation date."""
+    return sorted(notification, key=attrgetter("creation"), reverse=True)
+
+
+def create_message(
+    session: Session,
+    msg: str,
+    recipient_id: int,
+    link=None,
+) -> Message:
+    """Creates a new message."""
+    return create_model(
+        session,
+        Message,
+        body=msg,
+        recipient_id=recipient_id,
+        link=link,
+    )
+
+
+def get_all_messages(session: Session, recipient_id: int) -> List[Message]:
+    """Returns all messages."""
+    condition = Message.recipient_id == recipient_id
+    return session.query(Message).filter(condition).all()
+
+
+def get_all_invitations(session: Session, **param) -> List[Invitation]:
+    """Returns all invitations filter by param."""
+    try:
+        invitations = session.query(Invitation).filter_by(**param).all()
+    except SQLAlchemyError:
+        return []
+    else:
+        return invitations
+
+
```

Comment:
I love the short and intuitive way you create functions - short, readable, well documented with type annotations. Well done.